src/placement/jump_map_versions.c

```diff
@@ -15,24 +15,28 @@
 #include <isa-l.h>
 
 static bool
-is_new_added_dom(struct pool_component *comp)
+is_excluded_comp(struct pool_component *comp, uint32_t allow_version,
+		 enum layout_gen_mode gen_mode)
 {
-	return (comp->co_status == PO_COMP_ST_UP && comp->co_fseq <= 1) ||
-		comp->co_status == PO_COMP_ST_NEW;
-}
-
-static bool
-is_excluded_comp(struct pool_component *comp, bool exclude_new)
-{
-	/* All other non new added targets/ranks should not be excluded */
-	if (is_new_added_dom(comp) && exclude_new)
+	if (comp->co_status == PO_COMP_ST_NEW)
```

Comment:
NEW means the pool map does not know these target yet. so they can be ignored for layout generation.

All other states needs to be considered in jump hash, and those DOWN/DOWNOUT targets will remapped during layout generation,  otherwise it will cause data corruption.

is_comp_avaible means if the component is available for the layout, i.e. whether the shard can choose this target, no need remap.

is_excluded_comp() means if the component needs to be considered in this layout generation (jump hash process). so they are totally different.